Add kafka SASL version control to kafka_consumer

danielnelson opened the pull request:
Add manual control of SASL version for testing.  Before merging this support should also be added to the output.

related #6342

### Required for all PRs:

- [ ] Signed [CLA](https://influxdata.com/community/cla/).
- [ ] Associated README.md updated.
- [ ] Has appropriate unit tests.

danielnelson commented:
To use with Azure EventHub:
- Set `enable_tls = true`
- Set `sasl_version = 0`
- Patch Shopify/sarama with https://github.com/Shopify/sarama/pull/1542, this will hopefully not be required soon.

reimda commented on plugins/inputs/kafka_consumer/kafka_consumer.go:
```diff
@@ -158,20 +166,32 @@ func (k *KafkaConsumer) Init() error {
 		config.Net.TLS.Config = tlsConfig
+
+		// TLS is forced on if a custom tls.Config is set.
 		config.Net.TLS.Enable = true
```
If my config had custom TLS settings and also "enable_tls = false" I would expect the TLS config to be ignored and TLS not to be used.  This line would enable tls even though I explicitly said not to.

danielnelson replied:
I'll update the option to be a `*bool` so we can have the distinction between unset and `enable_tls = false`.

reimda commented on plugins/outputs/kafka/kafka.go:
```diff
@@ -258,20 +263,32 @@ func (k *Kafka) Connect() error {
 		config.Net.TLS.Config = tlsConfig
+
+		// TLS is forced on if a custom tls.Config is set.
 		config.Net.TLS.Enable = true
```
same here

danielnelson commented:
I added the EnableTLS option as a `*bool`, so that we have 3 states.  When set to false, TLS will not be used.  For backwards compatibility, if unset TLS will be enabled when the tlsconfig is modified.

I also added a warning message that will be displayed encouraging the use of `enable_tls` to control this setting, in 2.0 this option should be required to turn on TLS.  I'd like to start adding more warnings like this so that it will be easier to update a configuration before a major release.
